src/constructors/regular.c

```diff
@@ -420,6 +420,81 @@ igraph_error_t igraph_tree(igraph_t *graph, igraph_integer_t n, igraph_integer_t
     return IGRAPH_SUCCESS;
 }
 
+igraph_error_t igraph_tree_from_parent_vector(igraph_t *graph, igraph_vector_int_t* parents,
+                igraph_tree_mode_t type) {
+
+    igraph_vector_int_t edges = IGRAPH_VECTOR_NULL;
+    igraph_vector_int_t visited = IGRAPH_VECTOR_NULL;
+    igraph_integer_t idx = 0;
+    igraph_integer_t to = 1;
+
+    if (igraph_vector_int_empty(parents)) {
+        IGRAPH_ERROR("Parent vector is empty", IGRAPH_EINVAL);
+    }
+    if (type != IGRAPH_TREE_OUT && type != IGRAPH_TREE_IN &&  type != IGRAPH_TREE_UNDIRECTED) {
+        IGRAPH_ERROR("Invalid mode argument", IGRAPH_EINVMODE);
+    }
+
+    igraph_integer_t n = igraph_vector_int_size(parents);
+    IGRAPH_VECTOR_INT_INIT_FINALLY(&visited, n);
+    IGRAPH_VECTOR_INT_INIT_FINALLY(&edges, 2 * (n - 1));
+
+    // check if parents contains any cycles
+    igraph_integer_t child_node, parent_node, v;
+    for (int v=n-1; v>=0; --v) {
+      child_node = v;
```

Comment:
You sometimes have an indentation of 2 instead of 4. This makes it hard to see blocks.